src/wayland/shell/xdg/handlers/surface/toplevel.rs

```diff
@@ -206,6 +210,12 @@ pub fn send_toplevel_configure(resource: &xdg_toplevel::XdgToplevel, configure:
     };
     let serial = configure.serial;
 
+    // send bounds if requested
+    if send_bounds && resource.version() >= xdg_toplevel::EVT_CONFIGURE_BOUNDS_SINCE {
```

Comment:
Do we need an extra `send_bounds`, if `None` (or width=0 and height=0) mean the same thing?

Reply:
I added this to prevent excessive `configure_bounds` events. The spec says it must be sent initially and whenever it changes. Without that we would always send the bounds, I am not sure how a client will react on that, maybe it is not an issue at all.

Reply:
But the state is double buffered, no? Couldn't we just send the event whenever it changes?

Reply:
Oh now I see, the `pub fn send_toplevel_configure` was confusing me, I thought this was part of the public api. In this case `send_configure` does internally exactly what I would expect. Consider this solved.